docs/concepts/glossary.md

```diff
@@ -0,0 +1,168 @@
+In the context of Apache Flink, the term *parallel instance* is also frequently used to emphasize
+that multiple instances of the same [Operator](./glossary#operator) or
+[Function](./glossary#function) type are running in parallel.
+
+#### Flink Job
+
+A Flink Job is the runtime representation of a Flink program. A Flink Job can either be submitted
+to a long running [Flink Session Cluster](./glossary#flink-session-cluster) or it can be started as a
+self-contained [Flink Application Cluster](./glossary#flink-application-cluster).
+
+#### JobGraph
+
+see [Logical Graph](./glossary#logical-graph)
+
+#### Flink JobManager
+
+JobManagers are one of the components running in the
+[Flink JobManger Process](./glossary#flink-jobmanager-process). A JobManager is responsible for
+supervising the execution of the [Tasks](./glossary#task) of a single job.
+
+#### Logical Graph
+
+A logical graph is a directed graph describing the high-level logic of a stream processing program.
+The nodes are [Operators](./glossary#operator) and the edges indicate input/output-relationships or
+data streams or data sets.
+
+#### Managed State
+
+Managed State describes application state which has been registered with the framework. For
+Managed State, Apache Flink will take care about persistence and rescaling among other things.
+
+#### Flink JobManager Process
+
+The JobManager Process is the master of a [Flink Cluster](./glossary#flink-cluster). It is called
+*JobManager* for historical reasons, but actually contains three distinct components:
+Flink Resource Manager, Flink Dispatcher and one [Flink JobManager](./glossary#flink-jobmanager)
+per running [Flink Job](./glossary#flink-job).
+
+#### Operator
+
+Node of a [Logical Graph](./glossary#logical-graph). An Operator performs a certain operation,
+which is usually executed by a [Function](./glossary#function). Sources and Sinks are special
+Operators for data ingestion and data egress.
+
+#### Operator Chain
+
+An Operator Chain consists of two or more consecutive [Operators](./glossary#operator) without any
+repartitioning in between. Operators within the same Operation Chain forward records to each other
+directly without going through serialization or Flink's network stack.
+
+#### Partition
+
+A partition is an independent subset of the overall data stream or data set. A data stream or
+data set is divided into partitions by assigning each [record](./glossary#Record) to one or more
+partitions. Partitions of data streams or data sets are consumed by [Tasks](./glossary#task) during
+runtime. A transformation which changes the way a data stream or data set is partitioned is often
+called repartitioning.
+
+#### Physical Graph
+
+A physical graph is the result of translating a [Logical Graph](./glossary#logical-graph) for
+execution in a distributed runtime. The nodes are [Tasks](./glossary#task) and the edges indicate
+input/output-relationships or [partitions](./glossary#partition) of data streams or data sets.
```

Comment:
"physical graph" somehow implies this is the runtime graph of task instances on the nodes, but this is still only a data structure of how to deploy things, right?

Reply:
That's a good question and also related to your other comment below. I would have said that the "physical graph" indeed is the "runtime graph" of task instances. Of course it is just a model of the actual physical graph, but it models the execution during runtime.